src/legacy/ui/public/agg_types/controls/components/input_list.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, useEffect, Fragment } from 'react';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  htmlIdGenerator,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export interface InputListConfig {
+  defaultValue: {};
+  defaultEmptyValue: {};
+  validateClass: new (value: string) => object;
+  getModelValue(item: InputObject): {};
+  getModel(models: InputModel[], index: number, modelName?: string): InputModel | InputItem;
+  getRemoveBtnAriaLabel(model: InputModel): string;
+  onChangeFn(model: InputModel): InputObject;
+  hasInvalidValuesFn(model: InputModel): boolean;
+  renderInputRow(
+    model: InputModel,
+    index: number,
+    onChangeFn: (index: number, value: string, modelName?: string) => void
+  ): React.ReactNode;
+  validateModel(
+    validateFn: (value: string | undefined, modelObj: InputItem) => void,
+    object: InputObject,
+    model: InputModel
+  ): void;
+}
+interface InputModelBase {
+  id: string;
+  [key: string]: any;
+}
```

Comment:
This type definition is not providing much type safety because the `[key: string]: any` is so generic. Can you restrict it more?